modules/docs/lib/widgets/enhancedComponent.tsx

```diff
@@ -0,0 +1,200 @@
+import {Box} from '@workday/canvas-kit-react/layout';
+
+import {defaultJSDoc} from '../../docgen/docParser';
+import {EnhancedComponentValue} from '../../docgen/plugins/customTypes';
+import {MdxJSToJSX, MDX} from '../MDXElements';
+import {SymbolDoc} from '../SymbolDoc';
+import {PropertiesTable, registerWidget, Value} from '../Value';
+import {Heading, SymbolDialog} from '../widgetUtils';
+import * as types from '../../docgen/docTypes';
+
+const ParentComponentNameContext = React.createContext('');
+/**
+ * Used to keep track of JSDoc in subcomponent definitions.
+ *
+ * For example:
+ * ```tsx
+ * subComponents: {
+ *   // JSDoc goes here instead of on `TabsItem` for better dev experience
+ *   Item: TabsItem
+ * }
+ * ```
+ */
+const ParentComponentJSDocContext = React.createContext(defaultJSDoc);
+
+const fileNameToCategoryMap: Record<string, string[]> = {
+  Layout: [
+    'utils/layout',
+    'utils/gridArea',
+    'utils/gridItem',
+    'utils/flexItem',
+    'utils/position',
+    'utils/space',
+    'utils/stack',
+  ],
+  Color: ['utils/color', 'utils/background'],
+  Border: ['utils/border'],
+  Depth: ['utils/depth'],
+  Text: ['utils/text'],
+  Other: ['utils/other'],
+};
+
+registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta}) => {
+  const groups = groupProps(value.props);
+  const parentComponentName = React.useContext(ParentComponentNameContext);
+  const parentComponentJSDoc = React.useContext(ParentComponentJSDocContext);
+  const hideHeader = meta && meta.hideHeader === true;
+
+  const intro = (
+    <>
+      {!parentComponentName && !hideHeader && value.displayName ? (
+        <Heading>{value.displayName}</Heading>
+      ) : null}
+
+      <ParentComponentNameContext.Provider value="">
+        <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+          <MdxJSToJSX>{parentComponentJSDoc.description || doc?.description || ''}</MdxJSToJSX>
+        </ParentComponentJSDocContext.Provider>
+      </ParentComponentNameContext.Provider>
+    </>
+  );
+
+  // We don't want to re-document a subcomponent that is a container component with a different
+  // model. If we detect another container, only render the intro and a link to the container
+  // component
+  if (parentComponentName && value.componentType === 'container') {
+    return (
+      <>
+        {intro}
+        <MDX as="p">
+          This component references the{' '}
+          <ParentComponentNameContext.Provider value="">
+            <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+              <SymbolDialog value={{kind: 'symbol', name: value.displayName || ''}} />
+            </ParentComponentJSDocContext.Provider>
+          </ParentComponentNameContext.Provider>{' '}
+          component.
+        </MDX>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {intro}
+      {value.styleComponent ? (
+        <>
+          <Heading headingOffset={1}>Style Component</Heading>
```

Comment:
This `Style Component` section feels too terse to me:

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/146020/222032938-0c1e460d-0a78-475f-a11d-f6586fdd491b.png">

What do you think about renaming the section to `Layout Components` and adding a bit of explanation around the link to the underlying layout component? Something like this:

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/146020/222033059-d45f4fff-5ef5-4661-9a19-c612805d5ce9.png">

Reply:
I like it!